daemon/src/peer.rs

```diff
@@ -204,10 +204,13 @@ where
                         addrs_tx
                             .send(bound.listen_addrs())
                             .expect("subroutines is gone");
-                        let (_, run) = net::protocol::accept(bound, disco.clone().discover());
+                        let (close_endpoint, run) =
+                            net::protocol::accept(bound, disco.clone().discover());
+                        let close_endpoint = OnDrop::new(close_endpoint);
                         if let Err(e) = run.await {
                             tracing::error!(error = ?e, "accept error");
                         }
+                        drop(close_endpoint);
```

Comment:
For my educational purposes, why are these gymnastics required? Why can't we just call `close_endpoint` here?

Reply:
Alex Crichton calls this a "drop bomb" (or bomb drop?): the compiler cannot drop the value earlier. But dropping the future while we're at `run.await` won't run any code after that point.